scripts/chief_deploy.py

```diff
@@ -106,20 +106,19 @@ def update_info(ctx):
 @task
 def setup_dependencies(ctx):
     with ctx.lcd(settings.SRC_DIR):
-        # Creating a virtualenv tries to open virtualenv/bin/python for
-        # writing, but because virtualenv is using it, it fails.
-        # So we delete it and let virtualenv create a new one.
-        python = os.path.join(VENV_BIN, 'python')
-        python27 = os.path.join(VENV_BIN, 'python2.7')
-        ctx.local('rm -f %s' % python)
-        ctx.local('rm -f %s' % python27)
+        # Dearly beloved. We gather here to destroy this virtualenv in the
+        # hopes that out of the ashes will rise another new and beautiful
+        # virtualenv with no mistakes in it.
+        ctx.local('rm -rf %s' % settings.VENV_DIR)
         ctx.local('virtualenv-2.7 --no-site-packages %s' % settings.VENV_DIR)
 
         # Activate virtualenv to append to the correct path to $PATH.
         activate_env = os.path.join(VENV_BIN, 'activate_this.py')
         execfile(activate_env, dict(__file__=activate_env))
 
+        ctx.local('%s install --upgrade pip' % os.path.join(VENV_BIN, 'pip'))
```

Comment:
I think I would lock this down to a specific version of pip to prevent possible apocalyptic futures where we go to deploy only to discover some new and exciting t-retro virus created by some specific version of pip/peep and then we all die.